README.md

```diff
@@ -190,6 +191,46 @@ class TritonPythonModel:
     that is created must have "TritonPythonModel" as the class name.
     """
 
+    @staticmethod
+    def auto_complete_config(model_config):
+        """`auto_complete_config` is called only once when the server is started
+        with `--strict-model-config=false`. Implementing this function is optional.
+        A no implementation of `auto_complete_config` will do nothing. This function
+        can be used to set `max_batch_size`, `input` and `output` properties of the
+        model using `pb_utils.set_max_batch_size`, `pb_utils.add_input`, and
+        `pb_utils.add_output`. These properties will allow Triton to load the model
+        with minimal model configuration in absence of a configuration file. This
+        function returns the `pb_utils.ModelConfig` object with these properties.
+
+        Note: All the objects in this function will go out of scope after exiting.
+        Should not store any objects in this function.
+
+        Parameters
+        ----------
+        model_config : string
+          A JSON string containing the available model configuration. The users can
+          refer to this configuration when setting the properties for this model.
+
+        Returns
+        -------
+        pb_utils.ModelConfig
+          An object containing the max_batch_size, inputs and outputs properties.
+        """
+        auto_complete_model_config = pb_utils.ModelConfig()
```

Comment:
I was thinking that the model config in model.py should be built independently and should not rely on the configuration in config.pbtxt. I think this is also more like the model file we have in other backends. May I ask the reason behind that the users should build upon the configuration provided in config.pbtxt?

Reply:
They are kind of dependent even now. For example, if the user has specified an input in the config.pbtxt file we will be returning an error saying that the input has already been specified. The reason I'm requesting this change is that if we ever (not now) decide to support other fields in the model config, we can more easily explain that the config returned here (if it doesn't conflict with the config provided through other mechanisms) will be used as the final configuration. Right now, it looks like we are merging two separate configurations which can be a bit confusing.

Reply:
> This will also centralize all the error checking in the stub side instead of being spread in pb_stub.cc and python.cc
For error checking like [CheckModelConfigMismatch](https://github.com/triton-inference-server/python_backend/blob/krish-python/src/python.cc#L2570), do we want to put this in the stub side? I was thinking that this should be part of auto-complete.

> we can more easily explain that the config returned here (if it doesn't conflict with the config provided through other mechanisms) will be used as the final configuration.
For auto-complete logic, we only query the config from model.py when there is missing information in the config from config.pbtxt. If the configuration in model.py is used as the final configuration, which contains the excisting config and the newly added config, is the auto-complete part still needed?

Reply:
Got it! Thank you both for all the suggestion, will modify accordingly.